concent_api/api-integration-additional-verification-test.py

```diff
@@ -0,0 +1,359 @@
+        PROVIDER_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        create_client_auth_message(PROVIDER_PRIVATE_KEY, PROVIDER_PUBLIC_KEY, CONCENT_PUBLIC_KEY),
+        headers = {
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=200,
+        expected_message_type=message.concents.SubtaskResultsSettled.TYPE,
+        expected_content_type='application/octet-stream',
+    )
+
+
+@count_fails
+def test_case_2_test_for_resources_failure_reason(cluster_consts, cluster_url, test_id):
+    current_time = get_current_utc_timestamp()
+    (subtask_id, task_id) = get_task_id_and_subtask_id(test_id, 'existing_file')
+
+    file_content = task_id
+    file_size = len(file_content)
+    file_check_sum = 'sha1:' + hashlib.sha1(file_content.encode()).hexdigest()
+
+    api_request(
+        cluster_url,
+        'send',
+        PROVIDER_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        get_subtask_results_verify(
+            task_id,
+            subtask_id,
+            current_time,
+            cluster_consts,
+            reason=message.tasks.SubtaskResultsRejected.REASON.ConcentResourcesFailure,
+            report_computed_task_size=file_size,
+            report_computed_task_package_hash=file_check_sum,
+            task_to_compute_size=file_size,
+            task_to_compute_package_hash=file_check_sum,
+        ),
+        headers = {
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=200,
+        expected_message_type=message.concents.ServiceRefused.TYPE,
+        expected_content_type='application/octet-stream',
+    )
+
+
+@count_fails
+def test_case_3_test_for_invalid_time(cluster_consts, cluster_url, test_id):
+    current_time = get_current_utc_timestamp()
+    (subtask_id, task_id) = get_task_id_and_subtask_id(test_id, 'existing_file')
+
+    file_content = task_id
+    file_size = len(file_content)
+    file_check_sum = 'sha1:' + hashlib.sha1(file_content.encode()).hexdigest()
+
+    api_request(
+        cluster_url,
+        'send',
+        PROVIDER_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        get_subtask_results_verify(
+            task_id,
+            subtask_id,
+            current_time - 2 * cluster_consts.additional_verification_call_time,
+            cluster_consts,
+            reason=message.tasks.SubtaskResultsRejected.REASON.VerificationNegative,
+            report_computed_task_size=file_size,
+            report_computed_task_package_hash=file_check_sum,
+            task_to_compute_size=file_size,
+            task_to_compute_package_hash=file_check_sum,
+        ),
+        headers = {
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=200,
+        expected_message_type=message.concents.ServiceRefused.TYPE,
+        expected_content_type='application/octet-stream',
+    )
+
+
+@count_fails
+def test_case_4_test_for_duplicated_request(cluster_consts, cluster_url, test_id):
```

Comment:
I am not sure if we really need cluster test to check for duplication - it can be tested in django tests and does not require cluster (the processing chain is very short and it ends on `concent_api`)

Reply:
I think if we test it also on cluster will be better than if we would not test it there :)

Reply:
quite frankly, it does not give us much benefit, but it can stay there if you want ;)